### DISCUSSION TITLE
Adding custom actions on the elements in glsp editor

### DISCUSSION THREAD

**Discussion by Abhay0294** [eclipse-glsp/glsp#695] [Q&A]
My requirement is as follows:

**1.** When an user clicks on any element(for ex: Brew element in the picture) on the glsp editor he/she should get the arrow(red pointer in the picture) option to connect to other element.
**2.** User need not have to click on the toolpalette(edge option in the palette) for that arrow thing, instead when the user click on an element he/she should get the arrow option.

Requesting you to please let me if this can be achieved. If yes, please suggest how can I achieve this.

<img width="539" alt="glsp_editor" src="https://user-images.githubusercontent.com/57217189/177083907-d6049134-c58b-43e6-b34c-9befe6f4757f.PNG">
.

**Comment by tortmayr** [comment]
HI  @Abhay0294,

sorry for the late reply. 
Yes, your requirement can be implemented with GLSP. To achieve this you basically have two options

1. Customizing the existing [NodeCreationTool](https://github.com/eclipse-glsp/glsp-client/blob/2b83212065ba6b4d5e5579e81c61d2496cb75ce4/packages/client/src/features/tools/node-creation-tool.ts#L36) resp. the  [NodeCreationToolMouseListener](https://github.com/eclipse-glsp/glsp-client/blob/2b83212065ba6b4d5e5579e81c61d2496cb75ce4/packages/client/src/features/tools/node-creation-tool.ts#L71)
2. Implement an entirely new custom node creation tool, and configure it in the diagram DI config.

For an example on how to bind tools please check out the DI config of the [toolsModule](https://github.com/eclipse-glsp/glsp-client/blob/2b83212065ba6b4d5e5579e81c61d2496cb75ce4/packages/client/src/features/tools/di.config.ts#L37]).

**Reply by Abhay0294** [reply]
Thanks @tortmayr for the response, wanted to know what implementation I can include in the custom node creation tool, to achieve that implementation.
